controllers/grafana_controller.go

```diff
@@ -79,21 +76,27 @@ func (r *GrafanaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	metrics.GrafanaReconciles.WithLabelValues(grafana.Namespace, grafana.Name).Inc()
 
-	finished := true
-	stages := getInstallationStages()
-	nextStatus := grafana.Status.DeepCopy()
-	vars := &grafanav1beta1.OperatorReconcileVars{}
+	defer func() {
+		if err := r.Status().Update(ctx, grafana); err != nil {
+			log.Error(err, "updating status")
+		}
+	}()
 
 	if grafana.IsExternal() {
-		nextStatus.Stage = grafanav1beta1.OperatorStageComplete
-		nextStatus.StageStatus = grafanav1beta1.OperatorStageResultSuccess
-		nextStatus.AdminUrl = grafana.Spec.External.URL
-		v, err := r.getVersion(ctx, grafana)
+		grafana.Status.Stage = grafanav1beta1.OperatorStageComplete
+		grafana.Status.AdminUrl = grafana.Spec.External.URL
+		version, err := r.getVersion(ctx, grafana)
 		if err != nil {
-			log.Error(err, "failed to get version from external instance")
+			grafana.Status.Version = ""
+			grafana.Status.LastMessage = err.Error()
+			grafana.Status.StageStatus = grafanav1beta1.OperatorStageResultFailed
+			return ctrl.Result{}, fmt.Errorf("failed to get version from external instance: %w", err)
```

Comment:
I guess it would make sense to add a `RequeueDelay` here as well to accommodate temporary connectivity issues.